operator/duties/validatorregistration.go

```diff
@@ -36,10 +36,11 @@ func (h *ValidatorRegistrationHandler) HandleDuties(ctx context.Context) {
 		case <-ctx.Done():
 			return
 
-		case slot := <-h.ticker:
+		case <-h.ticker.Next():
 			shares := h.validatorController.GetOperatorShares()
-
+			slot := h.ticker.Slot()
```

Comment:
either that or make the slot return with Next() and then just 
`slot := <-h.ticker.Next()`

Reply:
After reading the code of `Next` I understand the logic since `Next`'s channel is actually waiting for the slot to arrive. we could still pack the logic to one func that will wait and return a chan with the slot. (it'll still be a chan so you could select on it.)